clippy_lints/src/manual_and_or.rs

```diff
@@ -0,0 +1,193 @@
+            ControlFlow::Break(())
+        } else {
+            ControlFlow::Continue(())
+        }
+    })
+    .is_some()
+}
+
+fn contains_or(cond: &Expr<'_>) -> bool {
+    for_each_expr(cond, |e| {
+        if let ExprKind::Binary(ref n, _, _) = e.kind {
+            if n.node == BinOpKind::Or {
+                ControlFlow::Break(())
+            } else {
+                ControlFlow::Continue(())
+            }
+        } else {
+            ControlFlow::Continue(())
+        }
+    })
+    .is_some()
+}
+
+fn check_and<'tcx>(cx: &LateContext<'tcx>, expr: &Expr<'tcx>, cond: &Expr<'tcx>, then: &Expr<'tcx>) {
+    if let Some(parent) = get_parent_expr(cx, expr) {
+        if let ExprKind::If(_, _, _) = parent.kind {
+            return;
+        }
+    }
+    if contains_or(cond) || contains_or(then) || fetch_bool_expr(then).is_some() {
+        return;
+    }
+    if match then.kind {
+        ExprKind::Block(block, _) => !block.stmts.is_empty(),
+        _ => false,
+    } {
+        return;
+    }
+
+    let applicability = Applicability::MachineApplicable;
+    let cond_snippet = cx
+        .sess()
+        .source_map()
+        .span_to_snippet(cond.span)
+        .unwrap_or_else(|_| "..".to_string());
+
+    let then_snippet = extract_final_expression_snippet(cx, then).unwrap_or_else(|| "..".to_string());
+    let suggestion = format!("{cond_snippet} && {then_snippet}");
+    span_lint_and_sugg(
+        cx,
+        MANUAL_AND,
+        expr.span,
+        "this `if`-then-`else` expression can be simplified with `&&`",
+        "try",
+        suggestion,
+        applicability,
+    );
+}
+
+fn check_or<'tcx>(cx: &LateContext<'tcx>, expr: &Expr<'tcx>, cond: &Expr<'tcx>, else_expr: &Expr<'tcx>) {
+    if matches!(else_expr.kind, ExprKind::If(..)) || fetch_bool_expr(else_expr).is_some() {
+        return;
+    }
+    if match else_expr.kind {
+        ExprKind::Block(block, _) => !block.stmts.is_empty(),
+        _ => false,
+    } {
+        return;
+    }
+
+    let applicability = Applicability::MachineApplicable;
+    let cond_snippet = cx
+        .sess()
+        .source_map()
+        .span_to_snippet(cond.span)
+        .unwrap_or_else(|_| "..".to_string());
+
+    let else_snippet = extract_final_expression_snippet(cx, else_expr).unwrap_or_else(|| "..".to_string());
+    let suggestion = format!("{cond_snippet} || {else_snippet}");
+    span_lint_and_sugg(
+        cx,
+        MANUAL_OR,
+        expr.span,
+        "this `if`-then-`else` expression can be simplified with `||`",
+        "try",
+        suggestion,
+        applicability,
+    );
+}
+
+impl<'tcx> LateLintPass<'tcx> for ManualAndOr {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &Expr<'tcx>) {
+        if let Some(higher::If {
+            cond,
+            then,
+            r#else: Some(else_expr),
+        }) = higher::If::hir(expr)
+        {
+            if contains_let(cond) {
+                return;
+            }
```

Comment:
I think this is not needed anymore? yes?

Reply:
That's true, it's fixed now!